test/support/utils_test_base.h

```diff
@@ -373,6 +373,16 @@ void update_data(TTestDataTransfer& helper, Args&& ...args)
     struct TestClassName
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
+#if TEST_DPCPP_BACKEND_PRESENT
+#define DEFINE_TEST_2(TestClassName, TemplateParams1, TemplateParams2)                     \
+    template <typename TestValueType, typename TemplateParams1, typename TemplateParams2>  \
+    struct TestClassName : TestUtils::test_base<TestValueType>
+#else
+#define DEFINE_TEST_2(TestClassName, TemplateParams1, TemplateParams2)                      \
+    template <typename TemplateParams1, typename TemplateParams2>                           \
```

Comment:
Please do not exclude first template param ```typename TestValueType``` in this case too:
- if we implement some class method outside the class, the declaration of class and template params should be the same.

Reply:
So depends on ```#if TEST_DPCPP_BACKEND_PRESENT``` we may use inheritance `from ``TestUtils::test_base<TestValueType>``` or don't use it.

Reply:
Maybe I am misunderstanding the suggestion, but it seems that `DEFINE_TEST_2` is written in alignment with the `DEFINE_TEST_1` and `DEFINE_TEST` macros above. The comment above these macros states to write them as:
```
// 1) define class as
//      template <typename TestValueType>
//      struct TestClassName : TestUtils::test_base<TestValueType>
// 2) define class as
//      struct TestClassName
```
Are you suggesting this structure be changed for `DEFINE_TEST_2`?

Reply:
@mmichel11 you are right, it's ok.